MdeModulePkg/Core/Dxe/Misc/MemoryProtection.c

```diff
@@ -1554,6 +1554,18 @@ ApplyMemoryProtectionPolicy (
   //   // newly added region of a type that does not require protection
   //   return EFI_SUCCESS;
   // }
+
+  // To catch the edge case where the attributes are not consistent across the range, get the
+  // attributes from the page table to see if they are consistent. If they are not consistent,
+  // GetMemoryAttributes() will return an error.
+  if (MemoryAttributeProtocol != NULL) {
+    if (!EFI_ERROR (MemoryAttributeProtocol->GetMemoryAttributes (MemoryAttributeProtocol, Memory, Length, &OldAttributes)) &&

```

Comment:
Nit: can you move the call to MemoryAttributeProtocol out of the if statement?

You are saving a EFI_STATUS local variable, but readability is better having it outside.

Reply:
For upstream deviations I usually try to optimize for a smaller footprint, and I still think this is quite readable. Thoughts?